libraries/eosiolib/contracts/eosio/key_value.hpp

```diff
@@ -0,0 +1,1051 @@
+            return cmp == -1;
+         }
+
+         bool operator<=(const iterator& b) const {
+            auto cmp = internal_use_do_not_use::kv_it_compare(itr, b.itr);
+            return cmp <= 0;
+         }
+
+         bool operator>(const iterator& b) const {
+            auto cmp = internal_use_do_not_use::kv_it_compare(itr, b.itr);
+            return cmp == 1;
+         }
+
+         bool operator>=(const iterator& b) const {
+            auto cmp = internal_use_do_not_use::kv_it_compare(itr, b.itr);
+            return cmp >= 0;
+         }
+
+      private:
+         eosio::name contract_name;
+         const kv_index* idx;
+
+         size_t data_size;
+
+         uint32_t itr;
+         kv_it_stat itr_stat;
+
+         key_type key() const {
+            return idx->get_key(value());
+         }
+      };
+
+   public:
+      eosio::name name;
+      eosio::name table_name;
+      eosio::name contract_name;
+
+      kv_table* tbl;
+
+      kv_index() = default;
+
+      template <typename KF>
+      kv_index(KF&& kf) {
+         key_function = [=](const T& t) {
+            return make_key(std::invoke(kf, &t));
+         };
+      }
+
+      /**
+       * Search for an existing object in a table by the index, using the given key.
+       * @ingroup keyvalue
+       *
+       * @tparam K - The type of the key. This will be auto-deduced by the key param.
+       *
+       * @param key - The key to search for.
+       * @return An iterator to the found object OR the `end` iterator if the given key was not found.
+       *
+       * Example:
+       *
+       * @code
+       * // This assumes the code from the class example.
+       *
+       *     void myaction() {
+       *       // add dan account to table           - see put example
+       *
+       *       auto itr = addresses.index.account_name.find("dan"_n);
+       *       eosio::check(itr != addresses.index.account_name.end(), "Couldn't get him.");
+       *
+       *       auto itr = addresses.index.full_name.find("Dan Larimer");
+       *       eosio::check(itr != addresses.index.full_name.end(), "Couldn't get him.");
+       *     }
+       * @endcode
+       */
+      template <typename K>
+      iterator find(K key) {
+         uint32_t value_size;
+
+         auto prefix = make_prefix(table_name, name);
+         auto t_key = table_key(prefix, make_key(key));
+
+         auto success = internal_use_do_not_use::kv_get(db, contract_name.value, t_key.data(), t_key.size(), value_size);
+         if (!success) {
+            return end();
+         }
+
+         uint32_t itr = internal_use_do_not_use::kv_it_create(db, contract_name.value, prefix.data(), prefix.size());
+
+         int32_t itr_stat = internal_use_do_not_use::kv_it_lower_bound(itr, t_key.data(), t_key.size());
+         auto cmp = internal_use_do_not_use::kv_it_key_compare(itr, t_key.data(), t_key.size());
+
+         eosio::check(cmp == 0, "This key does not exist in this iterator");
```

Comment:
What purpose does this comparison serve?  You already checked that the key exists using get and the table could not have changed since then.

Reply:
I restructured this code to remove the get check and only use kv_it_key_compare instead since my understanding of the code is that will be more performant.